doc/performance-testing.md

```diff
@@ -0,0 +1,44 @@
+# Performance testing
+
+Shields has some basic tooling available to help you get started with
+performance testing.
+
+## Benchmarking the badge generation
+
+Want to micro-benchmark a section of the code responsible for generating the
+static badges? Follow these two simple steps:
+
+1. Surround the code you want to time with `console.time` and `console.timeEnd`
+   statements. For example:
+
+```
+console.time('makeBadge')
+const svg = makeBadge(badgeData)
+console.timeEnd('makeBadge')
+```
+
+2. Run `npm run benchmark:badge` in your terminal. An average timing will
+   be displayed!
+
+If you want to change the number of iterations in the benchmark, you can modify
+the values specified by the `benchmark:badge` script in _package.json_. If
+you want to benchmark a specific code path not covered by the static badge, you
+can modify the badge URL in _scripts/benchmark-performance.js_.
+
+## Profiling the full code
+
+Want to have an overview of how the entire application is performing? Simply
+run `npm run profile:server` in your terminal. This will start the server
```

Comment:
I was slightly confused by this at first because it only starts the badge server - not the front-end too. Do you reckon its worth either explaining this in the docs, or also starting the front-end with `concurrently` like we do with `start`?
(don't have a strong preference on which one)

Reply:
I'd suggest _not_ starting the frontend, as it'll load a bunch of random static badges. I imagine you would use this in conjunction with a script that fires specific requests against localhost (like the current performance benchmark script).

Reply:
Good point :+1:  Lets just note it in the docs then

Reply:
The docs are hopefully clearer now!